### PR TITLE
updated search to only return granted documents

### LINKED ISSUE #1637: Apply security on the search
Currently the search does not apply the security attached to the current user when returning the results. This means that the search result also contains items, which the user is actually not allowed to see.

The search should therefore also consider the security settings of Sulu. The short paragraph in http://docs.sulu.io/en/latest/bundles/security.html#checking-security explains a bit how that works, although a piece of code would also be nice in that place.

There are two possiblities solving this:
* filter the search results in the controller
* additionally index the security information and add that to the search criteria

The first approach would be quite easy, but makes pagination almost impossible. Adding the information to the search index means that the security logic might have to be repeated (not sure if there is an elegant way around this).

### PR DESCRIPTION
**Pull request by danrot**
Merge with https://github.com/massiveart/MassiveSearchBundle/pull/65 and https://github.com/sulu-io/sulu-standard/pull/550

__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes
- [x] submit changes to the documentation
- [x] update URL manager to also work with callbacks for more complicated index names
- [x] fix search result icon and translation for webspaces
- [x] Don't use indexes parameters from configuration directly in controller
- [x] Remove key from webspace config and use filename instead
- [x] Exclude secured entities from search results
- [x] Do not index search results if they already have a permission
- [x] Create a good default search controller for the website

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #1637 
| BC breaks        | index tag in xml mapping has changed
| Documentation PR | https://github.com/sulu-io/sulu-docs/pull/129

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
         "imagine/imagine": "~0.6.1",
         "liip/theme-bundle": "1.3.*",
         "massive/search-bundle": "dev-develop",
+        "massive/search-bundle": "dev-feature/index-name-evaluation",
```
Comment (wachterjohannes) [inline composer.json]: don`t forget to remove this after merging https://github.com/massiveart/MassiveSearchBundle/pull/65!

File: src/Sulu/Bundle/MediaBundle/Search/EventListener/PermissionListener.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    public function __construct(
+        FileVersionMetaRepository $fileVersionMetaRepository,
+        SearchManagerInterface $searchManager
+    ) {
+        $this->fileVersionMetaRepository = $fileVersionMetaRepository;
+        $this->searchManager = $searchManager;
+    }
+
+    public function onPermissionUpdate(PermissionUpdateEvent $event)
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Search/EventListener/PermissionListener.php]: short description in phpdoc

File: src/Sulu/Bundle/TestBundle/Resources/dist/sulu.yml [hunk 3]
```diff
@@ -55,6 +55,16 @@ sulu_core:
     translations: ['de', 'en']
     fallback_locale: 'en'
 
+sulu_content:
+    search:
+        mapping:
+            Sulu\Bundle\ContentBundle\Document\PageDocument:
+                index: page
+                category: page
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/TestBundle/Resources/dist/sulu.yml]: i think we have removed cateogry?

**Comment by wachterjohannes** [comment]
@danrot this works for normal pages but when i search for the homepage i get following error:

image

**Comment by wachterjohannes** [comment]
same error i get when i search for a collection

**Comment by wachterjohannes** [comment]
@danrot i have added a missing locale condition in this PR (:

File: src/Sulu/Bundle/TestBundle/Resources/dist/sulu.yml (hunk 3 above)
Comment (danrot) [inline src/Sulu/Bundle/TestBundle/Resources/dist/sulu.yml]: You are right, will remove it.

**Comment by danrot** [comment]
@wachterjohannes There was an error in the account configuration, I think this caused all the troubles you've had.

**Pull request merged**